model_api/python/model_api/models/sam_models.py

```diff
@@ -0,0 +1,204 @@
+                        prompt.reshape(-1, 2, 2), inputs["orig_size"]
+                    )
+                    point_labels = np.array([2, 3], dtype=np.float32).reshape(-1, 2)
+                else:
+                    point_coords = self.apply_coords(
+                        prompt.reshape(-1, 1, 2), inputs["orig_size"]
+                    )
+                    point_labels = np.array([1], dtype=np.float32).reshape(-1, 1)
+
+                processed_prompts.append(
+                    {
+                        "point_coords": point_coords,
+                        "point_labels": point_labels,
+                        "mask_input": self.mask_input,
+                        "has_mask_input": self.has_mask_input,
+                        "orig_size": np.array(
+                            inputs["orig_size"], dtype=np.int64
+                        ).reshape(-1, 2),
+                        "label": label,
+                    },
+                )
+        return processed_prompts
+
+    def apply_coords(
+        self, coords: np.ndarray, orig_size: np.ndarray | list[int] | tuple[int, int]
+    ) -> np.ndarray:
+        """Process coords according to preprocessed image size using image meta."""
+        old_h, old_w = orig_size
+        new_h, new_w = self._get_preprocess_shape(old_h, old_w, self.image_size)
+        coords = deepcopy(coords).astype(np.float32)
+        coords[..., 0] = coords[..., 0] * (new_w / old_w)
+        coords[..., 1] = coords[..., 1] * (new_h / old_h)
+        return coords
+
+    def _get_preprocess_shape(
+        self, old_h: int, old_w: int, image_size: int
+    ) -> tuple[int, int]:
+        """Compute the output size given input size and target image size."""
+        scale = image_size / max(old_h, old_w)
+        new_h, new_w = old_h * scale, old_w * scale
+        new_w = int(new_w + 0.5)
+        new_h = int(new_h + 0.5)
+        return (new_h, new_w)
+
+    def _check_io_number(
+        self, number_of_inputs: int | tuple[int], number_of_outputs: int | tuple[int]
+    ) -> None:
+        pass
+
+    def _get_inputs(self) -> tuple[list[str], list[str]]:
+        """Get input layer name and shape."""
+        image_blob_names = list(self.inputs.keys())
+        image_info_blob_names: list = []
+        return image_blob_names, image_info_blob_names
+
+    def postprocess(
+        self, outputs: dict[str, np.ndarray], meta: dict[str, Any]
+    ) -> dict[str, np.ndarray]:
+        """Postprocess to convert soft prediction to hard prediction.
+
+        Args:
+            outputs (dict[str, np.ndarray]): The output of the model.
+            meta (dict[str, Any]): Contain label and original size.
+
+        Returns:
+            (dict[str, np.ndarray]): The postprocessed output of the model.
+        """
+        probability = np.clip(outputs["scores"], 0.0, 1.0)
+        hard_prediction = (
+            outputs[self.output_blob_name].squeeze(0) > self.mask_threshold
+        ).astype(np.uint8)
```

Comment:
```suggestion
        )
```
`soft_prediction = hard_prediction * probability.reshape(-1, 1, 1)` is fine masking with bool. OpenVINO Tensors also support bool: https://docs.openvino.ai/2022.3/api/ie_python_api/_autosummary/openvino.runtime.Type.html

Reply:
Updated